.github/workflows/test_pull_requests.yml

```diff
@@ -145,6 +145,23 @@ jobs:
           # it is a no-op if MIN_REQ is not set
           MIN_REQ: ${{ matrix.MIN_REQ }}
 
+      # The two following steps try to cache and speedup tox env creation on linux.
+      # in the first one we produce the date as value we can refer to,
+      # in order to change cache key every ~24h.
+      # in the second we restore cache from previous run.
+      # a step later on cull the files in the directory we cache to avoid
+      # going over the GitHub 5G limit.
+      - name: Get current date to flush cache every day
+        id: date
+        run: echo "::set-output name=date::$(date +'%Y-%m-%d')"
+      - name: Cache tox Virtualenv
+        uses: actions/cache@v2
+        if: runner.os == 'Linux'
+        env:
+          cache-name: cache-tox-${{ matrix.platform }}-${{ matrix.python }}-${{ matrix.toxenv || matrix.backend }}-${{ steps.date.outputs.date }}
+        with:
+          path: /tmp/.tox/
+          key: ${{ matrix.platform }}-${{ matrix.python }}-${{ matrix.toxenv || matrix.backend }}-${{ hashFiles('setup.cfg') }}-${{ hashFiles('tox.ini') }}-${{ steps.date.outputs.date }}
```

Comment:
@Carreau what's the benefit of hashing `setup.cfg` and `tox.ini` as part of the key? Is that in case those files change we want to clear the cache?

Reply:
Yes, that is to invalidate the cache, so that if you ever bump a requirements the cache get invalidated. 
Tox.ini contains some `pip uninstall` commands for example.

Those two files are I believe the only two with pinned dependencies.

I don't think it has much importance as the cache is invalidated every 24h, but it may help.

Reply:
Yep that makes sense, I think this is a good approach.